Calculator/CalculatorBrain.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  CalculatorBrain.swift
+//  Calculator
+//
+//  Created by Jane on 9/23/15.
+//  Copyright (c) 2015 Gunn High School. All rights reserved.
+//
+// Video 3 42:36
+
+import Foundation
+
+class CalculatorBrain
+{
+    private enum Op: Printable
+    {
+        case Operand(Double)
+        case UnaryOperation(String, Double -> Double)
+        case BinaryOperation(String, (Double, Double) -> Double)
+        case Variable(String)
+        
+        var description: String {
+            get {
+                switch self {
+                case .Operand(let operand):
+                    return "\(operand)"
+                case .UnaryOperation(let symbol, _):
+                    return symbol
+                case .BinaryOperation(let symbol, _):
+                    return symbol
+                case .Variable(let variable):
+                    return variable
+                }
+            }
+        }
+    }
+    
+    private var opStack = [Op]()
+    var variableValues = [String:Double]()
+    private var error: String?
+    
+    private var knownOps = [String:Op]()
+    
+    init() {
+        func learnOp(op: Op) {
+            knownOps[op.description] = op
+        }
+        learnOp(Op.BinaryOperation("×", *))
+        learnOp(Op.BinaryOperation("÷") { $1 / $0 })
+        learnOp(Op.BinaryOperation("−") { $1 - $0 })
+        learnOp(Op.BinaryOperation("+", +))
+        learnOp(Op.UnaryOperation("√", sqrt))
+    }
+    
+    var program: AnyObject { //guaranteed to be a PropertyList
+        get {
+            return opStack.map { $0.description }
+        }
+        set {
+            if let opSymbols = newValue as? Array<String> {
+                var newOpStack = [Op]()
```

Comment:
Consider replacing the next 6 lines with map closure. For example on how to use map: http://www.electricpeelsoftware.com/2014/08/12/map-filter-reduce.html